src/all/kavita/src/eu/kanade/tachiyomi/extension/all/kavita/Kavita.kt

```diff
@@ -1009,64 +1012,69 @@ class Kavita(private val suffix: String = "") : ConfigurableSource, UnmeteredSou
 
             manga.thumbnail_url = if (preferences.LastVolumeCover) {
                 try {
-                    try {
-                        Log.d(LOG_TAG, "Fetching volumes for series ${result.seriesId}")
-                        val volumesRequest = GET("$apiUrl/Series/volumes?seriesId=${result.seriesId}", headersBuilder().build())
-                        val volumes = client.newCall(volumesRequest).execute().parseAs<List<VolumeDto>>()
-
-                        Log.d(LOG_TAG, "Found ${volumes.size} volumes for series ${result.seriesId}")
-//                        volumes.forEachIndexed { index, volume ->
-//                            Log.d(
-//                                LOG_TAG,
-//                                "Volume #${index + 1}: " +
-//                                    "id=${volume.id}, " +
-//                                    "number=${volume.number}, " +
-//                                    "name=${volume.name}, " +
-//                                    "cover=${volume.coverImage.isNotBlank()}, " +
-//                                    "chapters=${volume.chapters.size}",
-//                            )
-//                        }
-
-                        // Find all volumes that have SingleFileVolume chapters and valid covers
-                        val validVolumes = volumes.filter { volume ->
+                    Log.d(LOG_TAG, "Fetching volumes for series ${result.seriesId}")
+                    val seriesId = result.seriesId ?: 0
+                    val volumes = client.newCall(
+                        GET("$apiUrl/Series/volumes?seriesId=$seriesId", headersBuilder().build()),
+                    ).execute().parseAs<List<VolumeDto>>()
+
+                    val libraryType = getLibraryType(seriesId)
+                    val isComicLibrary = libraryType == LibraryTypeEnum.Comic || libraryType == LibraryTypeEnum.ComicVine
+
+                    val coverCandidates = mutableListOf<Triple<String, Boolean, Float>>()
+
+                    for (volume in volumes) {
+                        if (isComicLibrary && volume.number == -100000) {
+                            // Issue: use chapter covers
+                            for (chapter in volume.chapters) {
+                                if (!chapter.coverImage.isNullOrBlank()) {
```

Comment:
Use inversion of control. Flip the if statement so you can continue and reduce nesting. 

```

if (chapter.coverImage.isNullOrBlank()) continue;

...

```